Common/TableProducer/trackPropagation.cxx

```diff
@@ -89,17 +90,22 @@ struct TrackPropagation {
     ccdb->setLocalObjectValidityChecking();
 
     lut = o2::base::MatLayerCylSet::rectifyPtrFromFile(ccdb->get<o2::base::MatLayerCylSet>(lutPath));
-
-    matCorr = o2::base::Propagator::MatCorrType::USEMatCorrLUT;
-
     if (!o2::base::GeometryManager::isGeometryLoaded()) {
       ccdb->get<TGeoManager>(geoPath);
-      constexpr long run3grp_timestamp = (1619781650000 + 1619781529000) / 2;
-      grpo = ccdb->getForTimeStamp<o2::parameters::GRPObject>(grpPath, run3grp_timestamp);
-      o2::base::Propagator::initFieldFromGRP(grpo);
-      o2::base::Propagator::Instance()->setMatLUT(lut);
     }
-    mVtx = ccdb->get<o2::dataformats::MeanVertexObject>(mVtxPath);
+  }
+
+  void initCCDB(aod::BCsWithTimestamps::iterator const& bc)
+  {
+    if (runNumber == bc.runNumber()) {
+      return;
+    }
+    grpo = ccdb->getForTimeStamp<o2::parameters::GRPObject>(grpPath, bc.timestamp());
+    LOGF(info, "Setting magnetic field to %d kG for run %d from its GRP CCDB object", grpo->getNominalL3Field(), bc.runNumber());
+    o2::base::Propagator::initFieldFromGRP(grpo);
+    o2::base::Propagator::Instance()->setMatLUT(lut);
+    mVtx = ccdb->getForTimeStamp<o2::dataformats::MeanVertexObject>(mVtxPath, bc.timestamp());
```

Comment:
Not familiar with this but is the validity of the mean vertex the whole run?

Reply:
I saw that currently it is zero but I mean at mean term

Reply:
Yes, see Ruben's comment "The MeanVertex is a run specific calibration object (should be constantly created in relatively short time intervals during the run) and should be used in the reconstruction for e.g. primary vertexing. " but this will anyway be replaced by the CCDB fetcher which will detect if the timestamp goes out of the objects validity.